### ISSUE TITLE
use webp

### ISSUE THREAD

**Issue by maxbachmann** [ToolboxBodensee/toolbox-webseite#233]
Now that most browsers support webp (all but IE and browsers on IOS) using webp more would be nice. We also still need more image sizes for different screen sizes so mobile devices only need smaller files.

**Comment by davidblumer** [comment]
(all but IE and browsers on iOS) - sounds like we shouldn't use bleeding edge technologies?

**Comment by maxbachmann** [comment]
well we would always support jpg aswell, but seriously you can´t wait until apple supports technologies they did not invent themselves. And Microsoft already supports it in edge

**Comment by DO1JLR** [comment]
Ich lasse mich gerne von neuer fancy technologie überzeugen, solange das für alte versionen keine einschränkungen bringt!

**Comment by davidblumer** [comment]
Is there a way to convert every existing image to webp-format? so we can just keep adding jpg / png images and lektor automatically generates them on build?

**Comment by DO1JLR** [comment]
So wie ich das verstanden habe, kann lektor selber (noch) nicht mit webp umgehen.

Allerdings scheint es mit Imagemagic relativ einfach *([suche: imagemagic jpeg to webp](https://medium.com/@andysolomon/converting-your-images-to-webp-with-imagemagick-a9c56cf6b579))* zu sein, aus gängigen bildvormaten webp zu machen.

Ich kann mir gut vorstellen, das auch über das [Makefile](https://github.com/ToolboxBodensee/toolbox-webseite/blob/61af444650c873d7d8370d6fc4470b84a0bf372c/Makefile) von [travic-ci.org](https://github.com/ToolboxBodensee/toolbox-webseite/blob/61af444650c873d7d8370d6fc4470b84a0bf372c/.travis.yml) beim bauen der Webseite erzeugt wird. So dass der ganze prozess eigentlich *"unsichtbar"* im hintergrund läuft und das einzige das man davon mitbekommt, dass man unter einigen modernen Browsern plötzlich noch kleinere Ladezeiten hat.

**Comment by maxbachmann** [comment]
yes the plan is to automatically convert the images using imagemagic, so the images can still be added as jpg

**Comment by davidblumer** [comment]
Is lektor capable to use image-source-sets so we are able to fallback to jpg?

**Comment by DO1JLR** [comment]
Yes, you can generate all kind of html, html5, css, javascript, xml or what ever you want - code with lektor. You just need to write it in the template file or create a new template for this. 

It is not that lektor is a magical react-native or nodejs app, with a advanced AI that will think what the user could mean if he drag and drop an image somewhere in the project folder.
lektor is based on jinja2 templates and will use them to generate --- what ever is written in the template.
And yes, not everythin like css, javascript or something else need to be an template. lektor support static files (assets) too.

**Comment by DO1JLR** [comment]
@maxbachmann Du hast das Issue als "WIP" getagged.

Irgendwelche Fortschritte.

**Comment by maxbachmann** [comment]
Bisher nicht. Habe jetzt gestern mal einen großen Teil der issues gelöst und werde mir das jetzt mal nochmal im Laufe der Woche anschauen

**Comment by maxbachmann** [comment]
Habe mal so ein wenig damit begonnen, aber da werde ich mich mal anders drum kümmern gibt gerade einfach zu viele andere Projekte/Baustellen die wichtiger sind

**Comment by DO1JLR** [comment]
sachmal @maxbachmann 
wann geht es hier denn weiter?
Ich denke, du hast jetzt gerade zeit!

**Comment by maxbachmann** [comment]
Ich habe zwar keine Ahnung wie du da drauf kommst. Aber nein hatte ich nicht ;)
